backend-api/src/functions/asyncActiveSession/tokenService/tokenVerifier.ts

```diff
@@ -0,0 +1,28 @@
+import { jwtVerify, JWTVerifyResult, KeyLike } from "jose";
+import { errorResult, Result, successResult } from "../../utils/result";
+
+export class TokenVerifier implements ITokenVerifier {
+  async verify(
+    jwt: string,
+    key: KeyLike | Uint8Array,
+  ): Promise<Result<JWTVerifyResult>> {
+    let result: JWTVerifyResult;
+    try {
+      result = await jwtVerify(jwt, key);
+    } catch (error) {
+      return errorResult({
+        errorMessage: `Error verifying token signature: ${error}`,
+        errorCategory: "SERVER_ERROR",
```

Comment:
This needs to be a client error, I believe. If we can't verify the signature then something is very likely wrong with the token.

Reply:
Yes! Makes sense. Thanks - this has been updated now